code/modules/reagents/dispenser/dispenser2.dm

```diff
@@ -2,6 +2,7 @@
 	name = "chemical dispenser"
 	icon = 'icons/obj/chemical.dmi'
 	icon_state = "dispenser"
+	var/icon_state_active = "dispenser_active"
```

Comment:
Maybe the standard dispenser should be a subtype, if we're going to have to redefine this as null on all children?

Reply:
I've already considered this, but this'd require changes to many places and files for little benefit. And this is not something that applies to all children, only those with a non-standard frame.